src/Voron/Impl/LowLevelTransaction.cs

```diff
@@ -803,14 +800,7 @@ internal void FreePage(long pageNumber)
                     _scratchPagesTable.Remove(pageNumber);
                 }
 
-                if (_dirtyPages.Remove(pageNumber) == false &&
-                    _dirtyOverflowPages.TryGetValue(pageNumber, out long numberOfOverflowPages))
-                {
-                    _dirtyOverflowPages.Remove(pageNumber);
-
-                    if (numberOfOverflowPages > 1) // prevent adding range which length is 0
-                        _dirtyOverflowPages.Add(pageNumber + 1, numberOfOverflowPages - 1); // change the range of the overflow page
-                }
+                _dirtyPages.Remove(pageNumber);
```

Comment:
Can we just call `DiscardScratchModificationOn` here?

Reply:
Yes, I belive that would be best